classes/plugins/Plugin.inc.php

```diff
@@ -108,6 +109,18 @@ function register($category, $path) {
 		}
 
 		HookRegistry::register ('Installer::postInstall', array($this, 'installFilters'));
+
+		// Register template paths
+		if ($this->getEnabled() && (!is_a($this, 'ThemePlugin') || $this->isActive())) {
```

Comment:
I think it's better not to add `ThemePlugin`-specific concerns to ancestor classes. I'd suggest adding a helper function to `Plugin` that, when called, registers the resource; then, in `ThemePlugin`, check if the theme is activated and if so call it. Other kinds of plugins that want to use this would call the helper function according to their own needs.